llvm/include/llvm/Transforms/Vectorize/SandboxVectorizer/SeedCollector.h

```diff
@@ -54,6 +54,10 @@ class SeedBundle {
     NumUnusedBits += Utils::getNumBits(I);
   }
 
+  virtual void insert(Instruction *I, ScalarEvolution &SE) {
+    assert("Subclasses must override this function.");
+  }
+
```

Comment:
`= 0`?

Reply:
I debated about that, but making the function pure virtual means that the top-level class can't be directly instantiated and therefore can't be tested independently, and most of the functionality is in this class, not the subclasses.

I suppose I could make the test subclass it with a stub implementation, but that seems like extra complexity for not much gain.

Reply:
`llvm_unreachable`?

Reply:
Done

Reply:
If you debate about `=0`, assert, and `llvm_unreachable`, how about introducing `SeedBundleBase` and `SeedBundle`? Then you can have the `=0`.

Reply:
I don't think that would solve the problem. Right now `insert` isn't pure virtual so that the tests for SeedBundle (which has a fair amount of code) can test an actual SeedBundle. If SeedBundleBase has a pure virtual function, it would have the same untestable problem--and all the useful code would be inside it. That's basically just renaming the class. There needs to be a stub somewhere so it can be tested independently.

I suppose I'll make it pure virtual, and then make the test itself locally instantiate a "SeedBundleForTest" with the function overridden to a stub that looks exactly like the one now. I personally don't think that is appreciably better than an error, but whatever.